src/consumers.rs

```diff
@@ -0,0 +1,58 @@
+use std::collections::VecDeque;
+use simulators::Packet;
+
+//Consumer has a VecDeque and its buffer size
+struct Consumer {
+    vector: VecDeque<Packet>,
+    buffer_size: u32,
+}
+
+impl Consumer {
+    fn new(buffer_size: u32) -> Consumer {
+        Consumer {
+            vector: VecDeque::new(),
+            buffer_size: buffer_size,
+        }
+    }
+
+    //If the consumer doesn't have a buffer size (if it's 0), it enques
+    //IF the consumer does have a buffer size, it enques when it's length is less then it's buffer size
+    fn enqueue(&mut self, packet: Packet) -> () {
+        if self.buffer_size == 0 {
+            self.vector.push_back(packet);
+        } else if self.vector.len() < (self.buffer_size as usize) { 
+            self.vector.push_back(packet);
+        }
+    }
+
+    fn dequeue(&mut self) -> Option<Packet> {
+        self.vector.pop_front()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_infinite_consumer() {
```

Comment:
Don't think these tests are necessary, fine with eliding them altogether (especially considering we're checking the internal `vector.length` directly).

Reply:
Aside: using `let mut packet` and reusing the `packet` variable isn't idiomatic AFAICT because we aren't really "mutating" packet, we're using it to refer to two separate things. It's more common to just inline the original term (so `infiniteConsumer.enqueue(Packet(1))`) or using `let packet` twice, shadowing the first.

Also snake casing.